irc/nickserv.go

```diff
@@ -586,20 +587,25 @@ func nsInfoHandler(server *Server, client *Client, command string, params []stri
 }
 
 func nsRegisterHandler(server *Server, client *Client, command string, params []string, rb *ResponseBuffer) {
-	// get params
-	account, email := params[0], params[1]
-	var passphrase string
-	if len(params) > 2 {
-		passphrase = params[2]
+	details := client.Details()
+	account := details.nick
+	passphrase := params[0]
+	var email string
+	if 1 < len(params) {
+		email = params[1]
 	}
 
 	certfp := client.certfp
-	if passphrase == "" && certfp == "" {
-		nsNotice(rb, client.t("You need to either supply a passphrase or be connected via TLS with a client cert"))
-		return
+	if passphrase == "*" {
+		if certfp == "" {
+			nsNotice(rb, client.t("You need to either supply a passphrase or be connected via TLS with a client cert"))
```

Comment:
let's change this message to: _"You must be connected with TLS to use a client certificate"_

Reply:
How about: "You must be connected via TLS with a client certificate to do this"?

Reply:
Hmm, let's split the difference: "You must be connected with TLS and a client certificate to do this", using 'connected via TLS' just feels a touch off